streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java

```diff
@@ -476,6 +476,7 @@ public synchronized StreamsBuilder addStateStore(final StoreBuilder builder) {
      * @throws TopologyException if the processor of state is already registered
      */
     @SuppressWarnings("unchecked")
+    @Deprecated
```

Comment:
We should extend the JavaDoc, with `@deprecated` annotation to, and point to the new method that should be used.